openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -175,6 +190,118 @@ def tag_content_object(
     return get_content_tags(object_key, taxonomy_id=taxonomy.id)
 
 
+def get_object_tree_with_objecttags(
+    content_key: ContentKey,
+    include_children: bool = True,
+) -> tuple[TaggedContent, TaxonomyDict]:
+    """
+    Returns the object with the tags associated with it. If include_children is True, then it will also include
+    the children of the object and their tags.
+    """
+
+    def _get_object_tags(content_key: ContentKey, include_children: bool) -> QuerySet[ObjectTag]:
+        """
+        Return the tags for the object and its children using a single db query.
+        """
+        content_key_str = str(content_key)
+        if not include_children:
+            return ObjectTag.objects.filter(
+                object_id=content_key_str,
+                tag__isnull=False,
+                tag__taxonomy__isnull=False,
+            ).select_related("tag__taxonomy")
+
+        # We use a block_id_prefix (i.e. the modified course id) to get the tags for the children of the Content
+        # (course) in a single db query.
+        # ToDo: Add support for other content types (like LibraryContent and LibraryBlock)
+        if isinstance(content_key, UsageKey):
+            course_key_str = str(content_key.course_key)
+            block_id_prefix = course_key_str.replace("course-v1:", "block-v1:", 1)
+        elif isinstance(content_key, CourseKey):
+            course_key_str = str(content_key)
+            block_id_prefix = str(content_key).replace("course-v1:", "block-v1:", 1)
+        else:
+            raise NotImplementedError(f"Invalid content_key: {type(content_key)} -> {content_key}")
+
+        return ObjectTag.objects.filter(
+            Q(object_id__startswith=block_id_prefix) | Q(object_id=course_key_str),
+            Q(tag__isnull=False, tag__taxonomy__isnull=False),
+        ).select_related("tag__taxonomy")
```

Comment:
I would prefer to use an API method here from `oel_tagging.api` but I guess there isn't one at the moment, so we can use this for now. Maybe just mention

```
# There is no API method in oel_tagging.api that does this yet,
# so for now we have to build the ORM query directly.
return ObjectTag.objects.filter(...
```

Reply:
Done: fabb72918cf6c4f53787b6728275c7f440a45ab6